apps/aot/src/app/events/[year]/leaderboard/_components/overall-leaderboard.tsx

```diff
@@ -1,14 +1,47 @@
 import { Prisma, prisma } from '@repo/db';
 import { LEADERBOARD_RANKING_LIMIT } from '../constants';
-import { DataTableLeaderboard } from '@repo/ui/components/data-table-leaderboard';
-import { overallLeaderboardColumns, type OverallLeaderboardEntry } from './columns';
 import { getNextAdventDay } from '~/utils/time-utils';
 import { redisClient } from '@repo/redis';
 import { getAotChallengeIdsSoFar } from '../getAotChallengeIds';
 
-export const dynamic = 'force-dynamic';
+interface OverallLeaderboardEntry {
+  score: number | bigint;
+  name: string;
+  image: string | null;
+}
+
+export async function getOverallTableData(leaderboard: OverallLeaderboardEntry[]) {
+  const userData = await getUserData(leaderboard);
+  const rows = leaderboard.map((r) => {
+    const user = userData.find((u) => u.name === r.name);
+    if (user === undefined) {
+      throw new Error(`Leaderboard entry for ${r.name} does not have a matching user record`);
+    }
+    return {
+      ...user,
+      ...r,
+      score: Number(r.score),
+    };
+  });
+  return rows;
+}
+async function getUserData(leaderboardEntries: OverallLeaderboardEntry[]) {
+  const userData = await prisma.user.findMany({
+    where: {
+      name: {
+        in: leaderboardEntries.map((e) => e.name),
+      },
+    },
+    select: {
+      name: true,
+      bio: true,
+      roles: true,
+    },
+  });
+  return userData;
+}
```

Comment:
Suggestion:

```suggestion
async function getUserData(leaderboardEntries: OverallLeaderboardEntry[]) {
  const userIds = leaderboardEntries.map((entry) => entry.userId);

  const userData = await prisma.user.findMany({
    where: {
      id: { in: userIds },
    },
    select: {
      id: true,
      bio: true,
      roles: true,
    },
  });

  const userMap = new Map(userData.map((user) => [user.id, user]));

  return userMap;
}
```


Now you can directly user this map as a reference only when you need to render this user's bio/roles etc.

Reply:
Is there a guarantee that the order of user data will be ===. order of userids ?

Reply:
It doesn't have to be in the same order! When you iterate through the leaderboard entries, you can lookup the info from the map!